clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkClient.java

```diff
@@ -256,11 +256,13 @@ public void poll(long timeout, long now, PollCondition pollCondition) {
      * nor will it execute any delayed tasks.
      */
     public void pollNoWakeup() {
-        disableWakeups();
-        try {
-            poll(0, time.milliseconds(), null);
-        } finally {
-            enableWakeups();
+        synchronized (this) {
```

Comment:
@hachikuji 

>  Do we actually have any cases currently where we invoke poll from a completion handler?

I couldn't find one (or any other obvious issue), so I'm good with either solution, just let me know if you'd like me to adjust to using a param I guess :)